### PR TITLE
Dark theme and TYPESCRIPT!

### PR DESCRIPTION
**Pull request by bmiddha**
- added webpack for frontend typescript
- ported frontend to typescript (now 100% TYPESCRIPT!)
- added handy npm scripts
- fixed negative graduation countdown
- linting
- removed eslint and added more tslint rules
- added an option to have a port env variable (example: `PORT=8081 npm run start`)

### CODE REVIEW

File: frontend/index.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 window.onload = function () {
-    const times = [];
+    const times: Exam[] = [];
     const now = moment();
 
     // Template creates divs with class "final" for each final
     // Filter out finals with dates in the past, and add the class "past",
     //  which will apply a style to hide the element
-    const finalCards = Array.from(document.getElementsByClassName("final")).filter((f) => {
+    const finalCards = Array.from(document.querySelectorAll(".final")).filter((f: HTMLDivElement) => {
```
Comment (wilt00) [inline frontend/index.ts]: Per [this StackOverflow answer](https://stackoverflow.com/a/43967953/5991492), `getElementsByClassName` is literally 100x faster than `querySelectorAll`

File: tslint.json [hunk 2]
```diff
@@ -1,32 +1,60 @@
...
+                "variable-declaration": "onespace"
+            }
+        ],
+        "no-internal-module": true,
+        "no-trailing-whitespace": true,
+        "no-null-keyword": true,
+        "prefer-const": true,
+        "jsdoc-format": true
     }
 }
```
Comment (wilt00) [inline tslint.json]: TSLint is being deprecated some time this year ([see here](https://github.com/palantir/tslint/issues/4534)). At some point should migrate this to ESLint.

File: views/index.mst [hunk 3]
```diff
@@ -12,7 +12,7 @@ html,body {
     font-size: 20px;
     font-family: 'Montserrat', sans-serif;
     text-align: center;
-    background-color: #4ABDAC;
+    background-color: #333333;
```
Comment (wilt00) [inline views/index.mst]: Wooo!

File: tslint.json (hunk 2 above)
Comment (wilt00) [reply tslint.json]: And now this repository is connected forever with that tslint issue. Nuts.

**Comment by bmiddha** [comment]
requested changes have been made:
- moved away from querySelector
- switched to eslint

**Pull request merged**